Notification type - notification -- repos and services

John-Emad opened the pull request.

John-Emad committed:
- Adding NotificationType repository and interface.
- Adding NotificationType service and interface.

John-Emad committed:
- Adding NotificationType repository and interface.
- Adding NotificationType service and interface.

John-Emad committed:
- Adding Notification repository and interface.
- Adding Notification service and interface.
- Adding Scoped DI of Notification repository and service.
- Adding Scoped DI of NotificationType repository and service.

John-Emad committed:
- Adding Notification repository and interface.
- Adding Notification service and interface.
- Adding Scoped DI of Notification repository and service.
- Adding Scoped DI of NotificationType repository and service.

omar98alaa commented on Eskon.Service/Interfaces/INotificationService.cs:
```diff
@@ -0,0 +1,30 @@
+        public Task<Notification> AddNewNotificationAsync(Notification notification);
+        #endregion
+
+        #region Read
```
For read services... Only `GetNotificationsForUser` and `GetNotificationById` are required for now... We'll implement anything else if required later on.

omar98alaa commented on Eskon.Service/Interfaces/INotificationService.cs:
```diff
@@ -0,0 +1,30 @@
+
+        #region Read
+        public Task<List<Notification>> GetAllNotificationsForSpecificNotificationTypeIdAsync(Guid notificationTypeId);
+        public Task<List<Notification>> GetAllNotificationsForSpecificRecieverIdAsync(Guid recieverId);
```
You shouldn't pass `Id` to a service... Pass the whole Receiver instead.

omar98alaa commented on Eskon.Service/Interfaces/INotificationService.cs:
```diff
@@ -0,0 +1,30 @@
+        public Task<List<Notification>> GetAllNotificationsForSpecificRedirectionIdAsync(Guid redirectionId);
+        #endregion
+
+        #region Update
```
Please add `SetNotificationAsRead` service.

omar98alaa commented on Eskon.Service/Interfaces/INotificationService.cs:
```diff
@@ -0,0 +1,30 @@
+        public Task UpdateNotification(Notification notification);
+        #endregion
+
+        #region Delete
```
Would be better to have a single delete service and decide how it is performed by calling the desired repo method (soft/hard delete)

omar98alaa commented on Eskon.Service/Interfaces/INotificationTypeService.cs:
```diff
@@ -0,0 +1,30 @@
+        public Task EditNotificationTypeAsync(NotificationType notificationType);
+        #endregion
+
+        #region Delete
```
Single delete service and decide in its implementation

omar98alaa reviewed (changes requested):
Good job... just minor changes. (commented)
Also, the branch is now rebased on the `UnitOfWork` commit.
You just have to integrate these services and repos with the new change.

John-Emad committed:
Rebasing to unit of work

John-Emad committed:
- Removing Delete operation and leave only soft deletes
- Adding SetNotificationAsRead.
- Editing `GetAllNotificationsForSpecificRecieverAsync` to take User as an input.
- Adding `INotificationService` , `INotificationTypeService` to unit of work service.
- Adding `INotificationTypeRepository` , `INotificationTypeRepository` to unit of work reposiroty.
- Editing the Controller name of `GetStudentByID` to `GetUserById`.

omar98alaa commented on Eskon.Service/Services/NotificationService.cs:
```diff
@@ -0,0 +1,67 @@
+        #region Update
+        public async Task UpdateNotification(Notification notification)
+        {
+            notification.UpdatedAt = DateTime.UtcNow;
```
No need to set update time... It is already configured to update timestamps on `SaveChanges`.

omar98alaa reviewed (changes requested):
Just a small fix

John-Emad committed:
Remove updateAt from NotificationService
